### PR TITLE
Fixes to profile images for fog/Google Cloud Storage

### PR DESCRIPTION
**Pull request by jywarren**
Follow-up to #9807 to try fixing profile images

See: https://publiclab.org/profile/liz to test, or https://stable.publiclab.org/profile/liz

fixes https://github.com/publiclab/plots2/issues/9957

### CODE REVIEW

**Comment by jywarren** [comment]
Pushed this to unstable to test...

**Comment by jywarren** [comment]
OK, so now we are forming the path to the image correctly -- `/public/system/users/photos/000/000/001/thumb/aseprite_templates.png` however all profiles point to the same image, for some reason!

**Comment by jywarren** [comment]
OK, trying `:id_partition` instead of `:id` in the `url:` setting...

**Comment by jywarren** [comment]
OK - new images seem to be OK. But the problem is that old images followed a different scheme and are stored in locations like this:

https://storage.googleapis.com/publiclab-production/public/system/profile/photos/554/thumb/Bristol-539-7.jpg

(ref https://web.archive.org/web/20200318121317im_/https://publiclab.org/system/profile/photos/554/thumb/Bristol-539-7.jpg)

https://console.cloud.google.com/storage/browser/_details/publiclab-production/public/system/profile/photos/554/original/Bristol-539-7.jpg?project=public-lab

**Comment by jywarren** [comment]
We can check: https://unstable.publiclab.org/notes/firas/03-13-2017/exhibition-and-mapping-workshops-in-bourj-al-shamali

**Comment by jywarren** [comment]
Tested this in GitPod and it should work. Testing on unstable now.

**Comment by jywarren** [comment]
OK! I believe this is working and ready to deploy!!

**Comment by jywarren** [comment]
system test error should disappear:

```
Minitest::UnexpectedError:         Capybara::ExpectationNotMet: expected to find css "div.alert-success" but there were no matches
            test/system/post_test.rb:253:in `block in <class:PostTest>'
```

**Comment by icarito** [comment]
I'm finding missing images in a different directory. So I think we might have a routing issue. E.g. 
https://storage.googleapis.com/publiclab-production/public/system/profile/photos/7/original/384404446-square.png but it's being searched at /public/system/users/photos/000/000/007/thumb/384404446-square.png

**Comment by jywarren** [comment]
Ah so in these URLs, it's `users` but it should be `profiles`?

File: app/models/user.rb [hunk 1]
```diff
@@ -38,9 +38,10 @@ module Frequency
     c.validates_format_of_login_field_options = { with: Authlogic::Regex::LOGIN, message: I18n.t('error_messages.login_invalid', default: "can only consist of alphabets, numbers, underscore '_', and hyphen '-'.") }
   end
 
-  has_attached_file :photo, styles: { thumb: '200x200#', medium: '500x500#', large: '800x800#' },
-                                    url: '/system/profile/photos/:id/:style/:basename.:extension'
-  #:path => ":rails_root/public/system/images/photos/:id/:style/:basename.:extension"
+  has_attached_file :photo,
+                    styles: { thumb: '200x200#', medium: '500x500#', large: '800x800#' },
+                    url: '/public/system/profile/photos/:id/:style/:basename.:extension',
+                    path: ':rails_root/public/system/public/system/profile/photos/:id/:style/:filename'
```
Comment (jywarren) [inline app/models/user.rb:44]: Oh wait, the other way around. So the images you're finding are at `users` but we are pointing to `profile`?

Comment (jywarren) [reply app/models/user.rb:44]: I had been working from this old image: https://storage.googleapis.com/publiclab-production/public/system/profile/photos/554/thumb/Bristol-539-7.jpg which currently works.

**Comment by jywarren** [comment]
Confirmed it worked! 

so, images are now pointed at https://unstable.publiclab.org/public/system/profile/photos/152/thumb/DSCN6508.JPG

so i remove "unstable." and it works https://publiclab.org/public/system/profile/photos/152/thumb/DSCN6508.JPG

i'll try uploading a new image just to final check

great, i believe it worked!

it uploaded to https://publiclab.org/public/system/profile/photos/1/medium/aseprite_templates.png

ok, let's merge and deploy

**Comment by jywarren** [comment]
It worked!!! https://publiclab.org/notes/firas/03-13-2017/exhibition-and-mapping-workshops-in-bourj-al-shamali

**Pull request merged**